src/main/python/docs/source/guide/algorithms.rst

```diff
@@ -26,18 +26,176 @@ Prerequisite:
 
 - :doc:`/getting_started/install`
 
+This example goes through an algorithm from the list of builtin algorithms that can be applied to a dataset.
+For simplicity the dataset used for this is `MNist <http://yann.lecun.com/exdb/mnist/>`_,
+since it is commonly known and explored.
+
+If one want to skip the explanation then the full script is available in to bottom of this page.
 
 Step 1: Get Dataset
 -------------------
 
-TODO
+Systemds provide builtin for downloading and setup of the mnist dataset.
+To setup this simply use::
 
-Step 2: Train model
--------------------
+    from systemds.examples.tutorials.mnist import DataManager
+    d = DataManager()
+    X = d.get_train_data()
```

Comment:
I get: 

`>>> X = d.get_train_data()

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.local/lib/python3.8/site-packages/systemds/examples/tutorials/mnist.py", line 56, in get_train_data
    self._get_data(self._train_data_url, self._train_data_loc)
  File "/.local/lib/python3.8/site-packages/systemds/examples/tutorials/mnist.py", line 102, in _get_data
    os.mkdir(folder)
FileNotFoundError: [Errno 2] No such file or directory: 'systemds/examples/tutorials/mnist'
`

Reply:
Should be fixed, The issue is that the mkdir method does not create nester folders, i replaced it with mkdirs.